pubnub/event_engine/manage_effects.py

```diff
@@ -129,7 +147,17 @@ def emit(self, effect: effects.PNEmittableEffect):
             self.emit_status(effect)
 
     def emit_message(self, effect: effects.EmitMessagesEffect):
-        self.pubnub._subscription_manager._listener_manager.announce_message('foo')
+        for message in effect.messages:
+            subscribe_message = SubscribeMessage().from_json(message)
+            pn_message_result = PNMessageResult(
+                message=subscribe_message.payload,
```

Comment:
Will it work with encrypted messages?

Reply:
yeah should work. I will add tests to cover it in upcoming changes